app/components/certificate/certificate-display.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  Flex,
+  Box,
+  HStack,
+  Text,
+  Heading,
+  Tooltip,
+  IconButton,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+  useToast,
+} from '@chakra-ui/react';
+import { DownloadIcon, CopyIcon } from '@chakra-ui/icons';
+import type { Certificate } from '@prisma/client';
+
+interface CertificateDisplayProps {
+  certificate: Certificate;
+  isPrivate: Boolean;
+}
+
+export default function CertificateDisplay(props: CertificateDisplayProps) {
+  const { isPrivate, certificate } = props;
+  const toast = useToast();
+  let description: string = '',
+    currentState: string = '',
+    key: string = '';
+
+  switch (isPrivate) {
```

Comment:
I don't think this makes sense.  The point of props is to pass down the values to be used.  You don't need to put any logic into this component, and can simply pass `currentState`, `key`, `description` down as props from the parent component.

Also, I would change the names:

* `currentState` is really `name` or `title`
* `key` is really `value`, which is either a `certificate` or a `private key` (i.e., `key` doesn't make sense as a common name)